src/org/mozilla/javascript/NativeObject.java

```diff
@@ -306,8 +306,16 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
                       else
                           break;
                   }
-                  if (gs != null)
+                  if (gs != null) {
+                      if (gs instanceof MemberBox) {
```

Comment:
Is there a way that we can do this without the instanceof check -- with that check, now NativeObject is aware of the implementation of MemberBox.

Reply:
I fear that was the way MemberBox was introduces (the instanceof  from your comment below was already there).
No idea how we can solve it without introducing many new interfaces and breaking the backward compatibility.

BTW: Simply count the occurrences of 'instanceof MemberBox' and compare it with 'instanceof Symbol' ;-)